flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/PartitionedFileWriteReadTest.java

```diff
@@ -116,6 +121,140 @@ void testWriteAndReadPartitionedFile() throws Exception {
         }
     }
 
+    @ParameterizedTest
+    @CsvSource({"true", "false"})
+    void testComputeReadablePosition(boolean broadcastRegion) throws IOException {
+        int numSubpartitions = 10;
+        int bufferSize = 1024;
+        int numBuffers = 1000;
+        int numRegions = 1;
+
+        List<Buffer>[] buffersWritten = new List[numSubpartitions];
+        List<Buffer>[] buffersRead = new List[numSubpartitions];
+        List<Tuple2<Long, Long>>[] regionStat = new List[numSubpartitions];
+        for (int subpartition = 0; subpartition < numSubpartitions; ++subpartition) {
+            buffersWritten[subpartition] = new ArrayList<>();
+            buffersRead[subpartition] = new ArrayList<>();
+            regionStat[subpartition] = new ArrayList<>();
+        }
+
+        int[] writeOrder = DataBufferTest.getRandomSubpartitionOrder(numSubpartitions);
```

Comment:
In the case of boundaries, we may need to test manually. For example: [0,1,2,3,....,sp-1]?